app/models/discussion_reader.rb

```diff
@@ -42,6 +42,11 @@ def volume
     end
   end
 
+  def discussion_reader_volume
+    # Crazy James says: necessary in order to get a string back from the volume enum, rather than an integer
+    self.class.volumes.invert[self[:volume]]
```

Comment:
Yeah... the only reason we need this is because we've already overloaded the `volume` method, which would return the thing we want. I had a thought about doing this:

```
def volume(reader_only: false)
  if reader_only
    super
  elsif persisted?
    # etc
   end
end
```
```
def discussion_reader_volume
  volume(reader_only: true)
end
```

...But decided against it because I'd just rather not touch existing code at all.